da/mock/mock.go

```diff
@@ -71,6 +73,9 @@ func (m *MockDataAvailabilityLayerClient) SubmitBlock(block *types.Block) da.Res
 func (m *MockDataAvailabilityLayerClient) CheckBlockAvailability(header *types.Header) da.ResultCheckBlock {
 	hash := header.Hash()
 	_, err := m.dalcKV.Get(hash[:])
+	if errors.Is(err, badger.ErrKeyNotFound) {
+		return da.ResultCheckBlock{DAResult: da.DAResult{Code: da.StatusSuccess}, DataAvailable: false}
+	}
 	if err != nil {
 		return da.ResultCheckBlock{DAResult: da.DAResult{Code: da.StatusSuccess}, DataAvailable: false}
```

Comment:
Now, we need to change also this line - any other error means actual erroneous situation. 
```suggestion
		return da.ResultCheckBlock{DAResult: da.DAResult{Code: da.StatusError, Message: err.Error()}, DataAvailable: false}
```